### PR TITLE
Baseline token generator/validator microservice

### PR DESCRIPTION
**Pull request by shumatepf**
First iteration of the IDVA token microservice. Documentation on how to locally setup and test are included in the readme.

This PR does not include github actions.

### CODE REVIEW

**Review by dzaslavskiy — CHANGES_REQUESTED** [review]

File: server/config.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# server/config.py
+
+from lib2to3.pytree import Base
```
[the remark this answers is not in the document]
Comment (shumatepf) [reply server/config.py:3]: Didn't notice this. Must've been autocomplete/import when I was trying to type `BaseConfig`.

File: server/auth/views.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                # check if token is expired
+                if token.is_expired():
+                    return Responses.expired()
+                else:
+                    return Responses.exist()
+        except Exception:
+            return Responses.unauthorized()
+
+
+class RefreshAPI(MethodView):
```
[the remark this answers is not in the document]
Comment (shumatepf) [reply server/auth/views.py:90]: Good point, it is more of an 'invoke' than a 'refresh'.

File: runtime.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+python-3.9.14
```
[the remark this answers is not in the document]
Comment (shumatepf) [reply runtime.txt:1]: Dependency versions are a bit dated (more so in the requirements file), but it was required for compatibility.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,128 @@
...
+### API Endpoints
+#### Token Registering
+Create a new token.
+
+`POST /auth/register`
+
+```
+Request body:
+{
+  "api_key": <api_key>,
```
Comment (dzaslavskiy) [inline README.md:57]: api key is in header now.

**Review by dzaslavskiy — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (shumatepf) [reply README.md]: Fixed.

**Review by dzaslavskiy — APPROVED** [review]

**Pull request merged**